.github/workflows/packages.yml

```diff
@@ -0,0 +1,105 @@
+name: Build all core packages
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/packages.yml'
+      - 'include/**'
+      - 'package/**'
+      - 'toolchain/**'
+      - 'config/**'
+jobs:
+  build:
+    name: Build packages with external toolchain
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: False
+      matrix:
+        include:
+          - name: malta-be
+            target: malta
+            subtarget: be
+          - name: x86-64
+            target: x86
+            subtarget: 64
+
+    container: registry.gitlab.com/openwrt/buildbot/buildworker-3.4.1
+
+    steps:
+      - name: Checkout master directory
+        uses: actions/checkout@v2
+        with:
+          path: openwrt
+
+      - name: Fix permission
+        run: |
+          chown -R buildbot:buildbot openwrt
+
+      - name: Initialization environment
+        run: |
+          TARGET=$(echo ${{ matrix.target }})
+          SUBTARGET=$(echo ${{ matrix.subtarget }})
+          echo "TARGET=$TARGET" >> "$GITHUB_ENV"
+          echo "SUBTARGET=$SUBTARGET" >> "$GITHUB_ENV"
+
+      - name: Parse toolchain file
+        working-directory: openwrt
+        run: |
+          TOOLCHAIN_FILE=$(curl "https://downloads.cdn.openwrt.org/snapshots/targets/${{ env.TARGET }}/${{ env.SUBTARGET }}/sha256sums" \
+            | sed -n -e 's/.*\(openwrt-toolchain.*\).tar.xz/\1/p')
+
+          echo "TOOLCHAIN_FILE=$TOOLCHAIN_FILE" >> "$GITHUB_ENV"
+
+      - name: Download external toolchain
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: |
+          wget -O - https://downloads.cdn.openwrt.org/snapshots/targets/${{ env.TARGET }}/${{ env.SUBTARGET }}/${TOOLCHAIN_FILE}.tar.xz \
+            | tar --xz -xf -
+
+      - name: Create configuration
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: |
+          echo CONFIG_ALL=y >> .config
+          echo CONFIG_ALL_KMODS=y >> .config
+          echo CONFIG_ALL_NONSHARED=y >> .config
+
+          ./scripts/ext-toolchain.sh \
+            --toolchain ${{ env.TOOLCHAIN_FILE }}/toolchain-* \
+            --overwrite-config \
+            --config ${{ env.TARGET }}/${{ env.SUBTARGET }}
+
+          make defconfig
+
+      - name: Show configuration
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: ./scripts/diffconfig.sh
+
+      - name: Build tools
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make tools/install -j$(nproc) BUILD_LOG=1
+
+      - name: Build toolchain
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make toolchain/install -j$(nproc) BUILD_LOG=1
+
+      - name: Build Kernel
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make target/compile -j$(nproc) BUILD_LOG=1
```

Comment:
We have our own CI step to test Kernel changes, do you want to rebuild them here to? I'd see if we can disable as much Kernel as possible and only check core packages without kmods.

Reply:
Agree that we can skip kernel build as we already have jobs that does the same.

Reply:
The kernel build does not test if we can package the kernel modules. For example if kmod misses a dependency or the .ko file does not exists the kernel test will not fail, but this will fail.

Reply:
imho we should find a way to compile kmods in the kernel build test. Currently we use make package/linux/compile that afaik package the .ko. Problem is that package like mac80211 require manual compilation as they are not handled by package/linux/compile